src/sage/rings/series_valuation.py

```diff
@@ -0,0 +1,169 @@
+"""
+Valuation for power series and Laurent series.
+"""
+
+# ****************************************************************************
+#       Copyright (C) 2025 Xavier Caruso <xavier@caruso.ovh>
+#
+#  Distributed under the terms of the GNU General Public License (GPL)
+#  as published by the Free Software Foundation; either version 2 of
+#  the License, or (at your option) any later version.
+#
+#                  https://www.gnu.org/licenses/
+# ****************************************************************************
+
+from sage.categories.fields import Fields
+from sage.structure.factory import UniqueFactory
+from sage.rings.valuation.valuation import DiscreteValuation
+
+class SeriesValuationFactory(UniqueFactory):
+    r"""
+    Create a valuation over a power series ring or a Laurent series ring.
+
+    INPUT:
+
+    - ``R`` -- the ring
+
+    EXAMPLES:
+
+        sage: R.<t> = PowerSeriesRing(QQ)
+        sage: v = R.valuation()
+        sage: v
+        t-adic valuation
+    """
+    def create_key_and_extra_args(self, R):
+        r"""
+        TESTS::
+
+            sage: R.<t> = LaurentSeriesRing(ZZ)
+            sage: R.valuation()
+            Traceback (most recent call last):
+            ...
+            NotImplementedError: valuation only implemented when the base ring is a field
+        """
+        if R.base_ring() not in Fields():
+            raise NotImplementedError("valuation only implemented when the base ring is a field")
+        return R, {}
+
+    def create_object(self, version, key, **extra_args):
+        r"""
+        Create a unique key identifying the valuation of ``R``.
+
+        EXAMPLES::
+
+            sage: R.<t> = PowerSeriesRing(GF(5))
+            sage: R.valuation()  # indirect doctest
+            t-adic valuation
+        """
+        from sage.rings.valuation.valuation_space import DiscretePseudoValuationSpace
+        parent = DiscretePseudoValuationSpace(key)
+        return parent.__make_element_class__(SeriesValuation_base)(parent)
+
+SeriesValuation = SeriesValuationFactory("sage.rings.series_valuation.SeriesValuation")
+
+
+class SeriesValuation_base(DiscreteValuation):
```

Comment:
Maybe `_generic` since there's nobody inheriting from this. Base sounds like abstract base to me (just a suggestion. Keep it if you want.)